### PR TITLE
Feature/no-class_name

### PR DESCRIPTION
**Pull request by markaren**
This PR removes the requirement of declaring `class_name` in user scripts. It finds the class name using regex. 

A bit unsure about if the binder example should be updated at this point. Which version does it run?

### CODE REVIEW

File: pythonfmu/builder.py [hunk 1]
```diff
@@ -36,6 +37,12 @@ def get_lib_extension() -> str:
     return platforms.get(platform.system(), "")
 
 
+def get_class_name(file_name: Path) -> str:
+    with open(str(file_name), 'r') as file:
+        data = file.read()
+        return re.search('class (.*)\(\s*Fmi2Slave\s*\)\s*:', data).group(1)
```
Comment (fcollonval) [inline pythonfmu/builder.py]:
```suggestion
        return re.search('^class (\w+)\(\s*Fmi2Slave\s*\)\s*:', data).group(1)
```

File: pythonfmu/pythonfmu-export/cpp/PySlaveInstance.cpp [hunk 2]
```diff
@@ -7,20 +7,36 @@
...
     std::ifstream infile(fileName);
     std::getline(infile, line);
     return line;
 }
 
+inline std::string findClassName(const std::string& fileName)
+{
+    std::string line;
+    std::ifstream infile(fileName);
+    std::string regexStr(R"(class (.*)\(\s*Fmi2Slave\s*\)\s*:)");
```
Comment (fcollonval) [inline pythonfmu/pythonfmu-export/cpp/PySlaveInstance.cpp]:
```suggestion
    std::string regexStr(R"(^class (\w+)\(\s*Fmi2Slave\s*\)\s*:)");
```

**Comment by fcollonval** [comment]
> A bit unsure about if the binder example should be updated at this point. Which version does it run?

Sorry I forgot to reply. The binder example is running on the `master` branch. So yes the notebook should be updated before merging this PR.

File: pythonfmu/builder.py (hunk 1 above)
Comment (markaren) [reply pythonfmu/builder.py]: `^` made it not match, but added your other suggestion.

**Pull request merged**